go/kbcrypto/nacl_sig_info.go

```diff
@@ -72,22 +76,33 @@ func (e UnhandledSignatureError) Error() string {
 }
 
 func (s NaclSigInfo) Verify() (*NaclSigningKeyPublic, error) {
+	return s.verifyWithPayload(s.Payload, false)
```

Comment:
It is just so that in the case where you pass in s.payload you skip comparing the thing to itself.  I think it is a worthwhile savings since a lot of this stuff is on the critical path